minject/metadata.py

```diff
@@ -142,6 +143,18 @@ def update_bindings(self, **bindings: DeferredAny) -> None:
         # TODO: 'lock' the bindings once added to the registry to make above note unnecessary
         self._bindings.update(bindings)
 
+    def update_async_context(self, is_async_context: bool) -> None:
```

Comment:
nit: this and the next function just update `is_async_context`. Should we make it a property or a getter/setter?